agent/doc/swdesign/README.md

```diff
@@ -29,31 +29,55 @@ Please note that the Ankaios Agent could also run on the same node as the Server
 
 The following diagram shows the structural view of the Ankaios Agent:
 
-![Overview](drawio/comp_overview.drawio.svg)
+![Overview](drawio/unit_overview.drawio.svg)
 
 ### AgentManager
 
-The AgentManager is the main component in the Ankaios Agent and is responsible, amongst others, for pushing Workloads to the runtime adapters and serving the Ankaios Control Interface by authorizing requests from the workloads to the Ankaios Server.
+The AgentManager is the entry component in the Ankaios agent and is responsible, amongst others, for receiving commands from the server and forwarding them to the RuntimeManager and for authorizing requests from the workloads to the Ankaios Server.
 
-For simplicity, the initial setup of the Ankaios Agent done in the main.rs is also counted as part of this unit.
+The initial setup of the Ankaios agent is done in the main.rs is also counted as part of this unit.
 
-### RuntimeAdapter
+### RuntimeManager
 
-The RuntimeAdapter provides common functionality used by all runtime adapters.
+The RuntimeManager holds a list of RuntimeFacades (more precisely a list of runtime connectors wrapped into a RuntimeFacade) and a list of running workloads. It is also responsible for the handling the update workload calls including the workload reuse and the logic of translating the added and deleted workload lists into commands to a RuntimeFacade or a WorkloadObject.
 
-### PodmanAdapter
+### RuntimeFacade
 
-The PodmanAdapter is a RuntimeAdapter for Podman. It is responsible for dealing with the podman Workloads (start, stop).
+The RuntimeFacade wraps some common actions shared between all runtime connectors, s.t. they don't need to be implemented multiple times. The RuntimeFacade is responsible for creating, resuming and replacing a WorkloadObject including the start of the workload control task. The RuntimeFacade is also responsible for providing functionality for deleting a workloads that do not have an internal WorkloadObject (found unneeded workloads started in a previous execution of the Ankaios agent).
 
-### PodmanWorkload
+### WorkloadObject
 
-The PodmanWorkload uses Podman REST API to process events from the PodmanAdapter. 
-The PodmanWorkload also checks periodically status of the podman workload and reports changes of the status. 
+A WorkloadObject represents a workload inside the Ankaios agent. It holds the control interface and the sender for the communication channel with the workload control task. The WorkloadObject is also responsible for managing the workload control task.
 
 ### ParameterStorage
 
 Stores information which the Agent gets from the Server. Currently the storage stores the workload status and mapping from workload name to runtime name.
```

Comment:
Is this still correct, I think the runtime name mapping is not stored anymore in the parameter storage?

In the code it is like the following: HashMap<AgentName, HashMap<WorkloadName, ExecutionState>>

```suggestion
Stores information which the Agent gets from the Server. Currently the storage stores the workload state of workloads for each agent.
```